pkg/analysis/utils/serialization/serialization_check.go

```diff
@@ -231,10 +237,33 @@ func (s *serializationCheck) handleFieldShouldBePointer(pass *analysis.Pass, fie
 	}
 }
 
-func (s *serializationCheck) handleFieldShouldNotBePointer(pass *analysis.Pass, field *ast.Field, fieldName string, isPointer bool, message string) {
+func (s *serializationCheck) handleFieldShouldNotBePointer(pass *analysis.Pass, field *ast.Field, fieldName string, isPointer bool, underlying ast.Expr, markersAccess markershelper.Markers, message string) {
 	if !isPointer {
 		return
 	}
 
+	// Check if this is a pointer-to-slice/map with explicit MinItems=0 or MinProperties=0
+	// In this case, the pointer is intentional to distinguish nil from empty
+	if hasExplicitZeroMinValidation(pass, field, underlying, markersAccess) {
+		return
+	}
+
 	reportShouldRemovePointer(pass, field, s.pointerPolicy, fieldName, message, fieldName)
 }
+
+// hasExplicitZeroMinValidation checks if a field has an explicit MinItems=0 or MinProperties=0 marker.
+// This indicates the developer intentionally wants to distinguish between nil and empty for slices/maps.
+func hasExplicitZeroMinValidation(pass *analysis.Pass, field *ast.Field, underlying ast.Expr, markersAccess markershelper.Markers) bool {
+	fieldMarkers := utils.TypeAwareMarkerCollectionForField(pass, markersAccess, field)
+
+	switch underlying.(type) {
+	case *ast.ArrayType:
+		// Check for explicit MinItems=0
+		return fieldMarkers.HasWithValue("kubebuilder:validation:MinItems=0")
```

Comment:
Makes this more complex, but, could we use the constants from the markers package here? Trying to avoid scattering strings everywhere so that we can trace exactly where marker constants are being used